### PR TITLE
Update code for free the blobs

### PR DESCRIPTION
**Pull request by kevaundray**

### CODE REVIEW

**Review by terencechain — APPROVED** [review]

**Comment by kevaundray** [comment]
Title updated as I also updated the other methods so that the PR now builds

File: core/types/transaction_marshalling.go [hunk 1]
```diff
@@ -54,7 +54,7 @@ type txJSON struct {
 	BlobVersionedHashes []common.Hash `json:"blobVersionedHashes,omitempty"`
 	Blobs               Blobs         `json:"blobs,omitempty"`
 	BlobKzgs            BlobKzgs      `json:"blobKzgs,omitempty"`
-	KzgAggregatedProof  KZGProof      `json:"kzgAggregatedProof,omitempty"`
+	Proofs              KZGProofs     `json:"proofs,omitempty"`
```
Comment (kevaundray) [inline core/types/transaction_marshalling.go:57]: I'm not sure if the json tag here is following some standard and it should be something like "kzgProofs"

File: core/types/transaction_test.go [hunk 2]
```diff
@@ -495,9 +495,9 @@ func TestTransactionCoding(t *testing.T) {
 			// This is the identity point serialised
 			var kzgProof KZGProof = [48]byte{192, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}
```
Comment (kevaundray) [inline core/types/transaction_test.go:496]: fwiw I don't think this belongs here -- the crypto could provide a global constant called `ZeroKZGProof` or something along those lines

Comment (roberto-bayardo) [reply core/types/transaction_test.go:496]: Agree

File: core/types/transaction_marshalling.go (hunk 1 above)
Comment (roberto-bayardo) [reply core/types/transaction_marshalling.go:57]: I think what you have is correct.

File: core/types/transaction_test.go (hunk 2 above)
Comment (kevaundray) [reply core/types/transaction_test.go:496]: created issue here: https://github.com/crate-crypto/go-proto-danksharding-crypto/issues/2

**Comment by roberto-bayardo** [comment]
This looks good, if it passes unit tests feel free to merge.

**Comment by kevaundray** [comment]
> This looks good, if it passes unit tests feel free to merge.

The tests were failing for :

```
FAIL    github.com/ethereum/go-ethereum/les/catalyst [build failed]
FAIL    github.com/ethereum/go-ethereum/eth/tracers/js [build failed]
FAIL    github.com/ethereum/go-ethereum/eth/tracers/logger [build failed]
```

The failures seem orthogonal to this change though

**Comment by kevaundray** [comment]
Merging as Roberto will look at the failing test cases -- orthogonal to this change

**Pull request merged**